### ISSUE TITLE
Cache account ids?

### ISSUE THREAD

**Issue by maltekliemann** [zeitgeistpm/zeitgeist#644]
Regarding `account_id()`, the treasury pallet's documentation says:

> This actually does computation. If you need to keep using it, then make sure you cache the value and only call this once. [https://github.com/paritytech/substrate/blob/7a28c62246406839b746af2201309d0ed9a3f526/frame/treasury/src/lib.rs#L437-L441]

We use this type of account generation in grand fashion. Should we cache market/pool accounts and trade chain storage for computation time?

**Comment by sea212** [comment]
Absolutely. Let's keep it a low priority for now because it is an optimizations that is not needed right now, but eventually will.
